x/evm/abci.go

```diff
@@ -542,13 +542,59 @@ func handleConfirmedEvents(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus,
 	}
 }
 
+func handleGeneralMessage(ctx sdk.Context, ck types.ChainKeeper, n types.Nexus, m types.MultisigKeeper, chain nexus.Chain, generalMessage nexus.GeneralMessage) error {
+	chainID := funcs.MustOk(ck.GetChainID(ctx))
+	keyID, ok := m.GetCurrentKeyID(ctx, chain.Name)
```

Comment:
This should have been checked when the general msg in created, so that we can simply panic here. There's no reason that a general msg should be created in the first place for an EVM chain without a key set. Based on this, I see no reason for the `Failed` status to even exist.

Reply:
I don't agree. `failed` status is for a chain state that is subject to change. Once there's a current key, it's never going to "unset". It's not subject to change.

Reply:
You can be in a scenario where an EVM chain is added, but it's gateway isn't set yet. call contract will succeed but panic here in every end blocker call. We shouldn't be adding evm specific checks in axelarnet module. We could add a chain validator similar to address validator in the future to refactor this, but not really in the scope for this week's release.

Reply:
Added some more checks. In general, the state could change between when the general message is created, and when it's being handled in the end blocker, at the very least due to being delayed by rate limiting. So we have to retain the failed status for that rare but possible scenario. In the future, we can add validation hooks to the nexus to do more checks in the message handler, to reduce the chance of failing in the end blocker.

Reply:
@milapsheth this check is NOT evm specific. Any non-cosmos chain would require a key to be set.